### PR TITLE
feat: use user tags to check PSA auth

### PR DESCRIPTION
**Pull request by alexandrastoica**
Removes the old `pinning_authorization` table and uses user tags instead

### CODE REVIEW

File: packages/api/test/fixtures/init-data.sql [hunk 1]
```diff
@@ -81,11 +81,10 @@ VALUES (
   1669394359626000000
 );
 
--- user 'test-pinning' is authorized
-INSERT INTO pinning_authorization (user_id)
-VALUES (4);
-INSERT INTO pinning_authorization (user_id)
-VALUES (5);
+-- -- user 'test-pinning' is authorized
```
Comment (flea89) [inline packages/api/test/fixtures/init-data.sql:84]: nit: double comment -- --

File: packages/api/test/fixtures/init-data.sql [hunk 2]
```diff
@@ -81,11 +81,10 @@ VALUES (
...
+INSERT INTO public.user_tag (user_id, tag, value, value_type, reason)
+VALUES  (4, 'PSA_ENABLED', true, 'boolean', 'test'),
+        (5, 'PSA_ENABLED', true, 'boolean', 'test');
```
Comment (trigramdev9) [inline packages/api/test/fixtures/init-data.sql:87]: please pull the latest.  `value_type` has been removed.

File: packages/db/postgres/pg-rest-api-types.d.ts [hunk 3]
```diff
@@ -1906,6 +2068,24 @@ export interface definitions {
...
+     * Note:
+     * This is a Primary Key.<pk/>
+     */
+    id: number;
+    /**
+     * Note:
+     * This is a Foreign Key to `user.id`.<fk table='user' column='id'/>
+     */
+    user_id: number;
+    tag: "ACCOUNT_ENABLED" | "PSA_ENABLED" | "STORAGE_LIMIT_BYTES";
```
Comment (trigramdev9) [inline packages/db/postgres/pg-rest-api-types.d.ts:2082]: please pull latest.  these have been renamed.

File: packages/db/index.js [hunk 4]
```diff
@@ -138,16 +138,18 @@ export class DBClient {
...
    */
   async isPinningAuthorized (userId) {
-    const { error, count } = await this._client
-      .from('pinning_authorization')
-      .select('id', { count: 'exact' })
+    const { count, error } = await this._client
+      .from('user_tag')
+      .select('value', { count: 'exact' })
       .eq('user_id', userId)
+      .eq('tag', 'PSA_ENABLED')
```
Comment (trigramdev9) [inline packages/db/index.js:151]: please pull latest.  this has been renamed to `HasPsaAccess`

**Review by trigramdev9 — CHANGES_REQUESTED** [review]

**Comment by alexandrastoica** [comment]
SQL for migrating the existing data from `pinning_authorization` to `user_tag`

```
BEGIN TRANSACTION;
INSERT INTO public.user_tag
(
    user_id,
    tag,
    value,
    reason,
	inserted_at
)
SELECT
     user_id,
     'HasPsaAccess' as tag,
     'true' as value,
     'Approved access' as reason,
	 inserted_at
 FROM public.pinning_authorization
 WHERE deleted_at IS NULL;
 
INSERT INTO public.user_tag
(
    user_id,
    tag,
    value,
    reason,
	inserted_at,
	deleted_at
)
SELECT
     user_id,
     'HasPsaAccess' as tag,
     'false' as value,
     'Revoked access' as reason,
	 inserted_at,
	 deleted_at
 FROM public.pinning_authorization
 WHERE deleted_at IS NOT NULL;

DROP TABLE public.pinning_authorization;
COMMIT;
```

**Comment by alanshaw** [comment]
@alexandrastoica could you add the migration SQL to here please? https://github.com/web3-storage/web3.storage/tree/main/packages/db/postgres/migrations

**Comment by flea89** [comment]
@alanshaw thanks for pointing that out! We noticed this morning that folder was created 🎉 

While I understand this is not the full, polished solution for migrations, (since we're still missing the automation side of it) I do believe moving migrations scripts to version control is a great improvement to what we're doing now. 
So big plus and thanks 🙏  ! 

I'm wondering if we should be adding some documentation to make sure developers know what's expected when altering schemas?

Do we have an issue to track the more holistic DB migration epic/approach?

**Comment by alexandrastoica** [comment]
@alanshaw added the migration file. Let me know if this is ok to merge. Thanks!

**Review by alanshaw — APPROVED** [review]
LGTM

**Comment by vasco-santos** [comment]
Runing SQL migration for deploy got:

```
web3-storage-prod::HEROKU_POSTGRESQL_AQUA=> INSERT INTO public.user_tag
(
    user_id,
    tag,
    value,
    reason,
    inserted_at
)
SELECT
    user_id,
    'HasPsaAccess' as tag,
    'true' as value,
    'Approved access' as reason,
    inserted_at
 FROM public.pinning_authorization
 WHERE deleted_at IS NULL;
ERROR:  invalid input value for enum user_tag_type: "HasPsaAccess"
LINE 11:     'HasPsaAccess' as tag,
```
Looking into it

**Comment by flea89** [comment]
@vasco-santos the user_tag enum was created as part of a separate PR from trigram folks, it's possible they haven't created a migration to create the ENUM?
```
-- User tags are associated to a user for the purpose of granting/restricting them
-- in the application.
CREATE TYPE user_tag_type AS ENUM
(
  'HasAccountRestriction',
  'HasPsaAccess',
  'StorageLimitBytes'
);
```

**Comment by trigramdev9** [comment]
> @flea89 thanks, you were right! The type was outdated and it was merged without updating the type. We need to improve communication on DB changes

@vasco-santos i'll take ownership for this in this instance.  if you recall the conversations we were having a few weeks ago about naming...  i meant to follow up once we were all in agreement and the conversations had died down to avoid unnecessary DB updates in prod, but i forgot to do so.

**Pull request merged**